wgpu-core/src/command/compute.rs

```diff
@@ -251,10 +251,16 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                     // Rebind resources
                     if state.binder.pipeline_layout_id != Some(pipeline.layout_id.value) {
                         let pipeline_layout = &pipeline_layout_guard[pipeline.layout_id.value];
-                        state.binder.pipeline_layout_id = Some(pipeline.layout_id.value);
-                        state
+
+                        let old_pipeline_layout_id = state
                             .binder
-                            .reset_expectations(pipeline_layout.bind_group_layout_ids.len());
+                            .pipeline_layout_id
+                            .replace(pipeline.layout_id.value);
+                        let old_pipeline_layout =
```

Comment:
would it be simpler to just follow up with a `map` here (instead of `old_pipeline_layout_id` intermediate)?

Reply:
The reason I went for a temporary here is because replace isn't just a chain, it's the operation setting the new value of `pipeline_layout_id` so to call attention to it, it gets it's own line.